src/1-theory/10-async.adoc

```diff
@@ -0,0 +1,407 @@
+HTTP ports 80 and 443 as well as to support HTTP proxies and intermediaries"
+thus making it compatible with the HTTP protocol. To achieve compatibility, the
+WebSocket handshake uses the HTTP Upgrade header to change from the HTTP
+protocol to the WebSocket protocol.
+----
+
+Basically whatever client and server side code you are using will do the same
+thing: the server provides a WebSocket URL to the client, and they connect, and
+messages can then go either way. Just like HTTP has `http://` and `https://`,
+there is `ws://` and `wss://`.
+
+[source,javascript]
+____
+// Create WebSocket connection
+const socket = new WebSocket('ws://localhost:8080/cakes/abc123');
+
+// Connection opened
+socket.addEventListener('open', function (event) {
+    socket.send('Hello Server!');
+});
+
+// Listen for messages
+socket.addEventListener('message', function (event) {
+    console.log('update RE that cake ', event.data);
+});
+____
+
+The fact that HTTP and WS are quite similar in many ways, and that they are
+compatible in general, make it really easy to add WebSocket's into your HTTP API
+by using a web server that has it built in. MDN maintain
+https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API#Tools[a list of
+WebSocket tools] for folks interested in giving it a try, including web servers
+that support them.
+
+You can punt figuring out how to run a WebSocket server entirely, but using
+hosted solutions like https://pusher.com/[Pusher] and
+https://www.pubnub.com/websockets/[PubNub].
+
+=== Message Brokers & Job Queues
+
+One important component here is the server being able to "do stuff later",
+instead of the client twiddling their thumbs waiting for updates. How exactly do
+we do that? Generally the idea is to push an event into a job queue, and then
+some sort of worker process will pick up that message and "work on the job".
+
+Sometimes this is done with tools based on AMQP or MQTT, two protocols designed
+for "Message Brokering" in general. That is a whole topic by itself, and job
+queues are just one of many possible use cases that message brokers can cover.
+These tools run their own server which acts as a "broker", and they send the
+messages off to "subscribers" who can get things done.
+
+A more simplistic approach for job queues it to use tooling like Sidekiq (a
+popular Ruby tool for handling job queues), or the more recent polyglot version
+Faktory. These sorts of solutions are popular because they just run off of Redis
+as the queue, and often run in the same codebase just on different threads. This
+makes the setup a bit less confusing for many.
+
+Whatever message broker / job worker / do it later approach you use, you need to
+let  the client know that work is being done in the background. For a HTTP API
+this is usually done with a `201 Created`, or `202 Accepted` response code,
+which also lets the client know they should look for some sort of link to get
+their updates.
```

Comment:
Is the link given in the response body? Worth mentioning how you can get the link?